crates/syn-solidity/src/yul/lit/mod.rs

```diff
@@ -0,0 +1,85 @@
+use crate::{kw, LitHexStr, LitStr, Spanned};
+
+use proc_macro2::Span;
+use std::fmt;
+use syn::{
+    parse::{Parse, ParseStream, Result},
+    LitBool, LitInt,
+};
+
+mod hex_num;
+pub use hex_num::YulHexNum;
+
+/// Yul literals e.g. 0x123, 42 or "abc".
+///
+/// Solidity Reference:
+/// <https://docs.soliditylang.org/en/latest/grammar.html#a4.SolidityParser.yulLiteral>
+#[derive(Clone)]
+pub enum YulLit {
```

Comment:
These are similar enough to Rust and Solidity that you can just use `crate::Lit`